### ISSUE TITLE
Android Talk Back not targeting Custom Header Center in Modal

### ISSUE THREAD

**Issue by gp3gp3gp3** [software-mansion/react-native-screens#998]
## Description

When using the `headerCenter` property on a `Stack.Screen`, Android's TalkBack screen reader is unable to target the center component.  The `headerLeft` custom component is able to use the `accessible` and `accessibilityLabel` like regular views, but only the `headerCenter` completely ignores the screen reader.  When using the `createStackNavigator` this does not occur, only when using `createNativeStackNavigator`

## Steps To Reproduce

1. Declare your stack with `import { createNativeStackNavigator } from 'react-native-screens/native-stack'`
2. Create a modal with `fullScreenModal` or any other kind.
3. Set a `headerCenter` component, a regular view with some text and an `accessibilityLabel`.
4. Note that Android Talk Back will skip over the header center component completely.

### Expected behavior
Screen reader should target text on screen
### Actual behavior
Screen reader ignores
## Snack or minimal code example
```
import { createNativeStackNavigator } from 'react-native-screens/native-stack'

const RootStack = createNativeStackNavigator()

<RootStack.Navigator>
  <RootStackScreen
    options={{
      headerCenter: () => <Text accessible={true} accessibilityRole="heading">Hello</Text>
    }}
    stackPresentation: 'fullScreenModal'
    component={MyModal}
  />
</RootStack.Navigator>
```

## Package versions

- React: 17.0.1
- React Native: 0.64.1
- React Native Screens:  3.3.0

**Comment by Ubax** [comment]
@gp3gp3gp3 Could you test TestsExample app on this branch https://github.com/software-mansion/react-native-screens/tree/%40ubax/android-accessibility. 
I tried to reproduce the issue there, but it works for me on Android emulator. If it does not work could you provide a code in TestsExample app?
